src/frontend/screens/Invites/InviteReceived.tsx

```diff
@@ -0,0 +1,149 @@
+import {BottomSheetWrapper} from '../../sharedComponents/BottomSheetWrapper';
+import InviteIcon from '../../images/AddPersonCircle.svg';
+import {StyleSheet, View} from 'react-native';
+import {LIGHT_GREY} from '../../lib/styles';
+import {PrimaryButton, SecondaryButton} from '../../sharedComponents/Buttons';
+import {defineMessages, useIntl} from 'react-intl';
+import {NativeNavigationComponent} from '../../sharedTypes/navigation';
+import {
+  useAcceptInvite,
+  useRejectInvite,
+  useSingleInvite,
+  useManyProjects,
+} from '@comapeo/core-react';
+import {HeaderText} from '../../sharedComponents/Text/HeaderText';
+import {BodyText} from '../../sharedComponents/Text/BodyText';
+import {UIActivityIndicator} from 'react-native-indicators';
+import {useActiveProjectIdActions} from '../../contexts/ActiveProjectIdStoreContext';
+import * as Sentry from '@sentry/react-native';
+import {useListenToInviteStateUpdate} from '../../hooks/useListenToInviteStateUpdate';
+
+const m = defineMessages({
+  navTitle: {
+    id: 'screens.InviteReceived.navTitle',
+    defaultMessage: 'Invite Received',
+  },
+  acceptInvite: {
+    id: 'screens.InviteReceived.acceptInvite',
+    defaultMessage: 'Accept Invite',
+  },
+  declineInvite: {
+    id: 'screens.InviteReceived.declineInvite',
+    defaultMessage: 'Decline Invite',
+  },
+  joinProject: {
+    id: 'screens.InviteReceived.joinProject',
+    defaultMessage: 'Join {projectName}',
+  },
+  invitedToJoin: {
+    id: 'screens.InviteReceived.invitedToJoin',
+    defaultMessage: "You've been invited to join {projectName}",
+  },
+});
+
+export const InviteReceived: NativeNavigationComponent<'InviteReceived'> = ({
+  route,
+  navigation,
+}) => {
+  const {formatMessage} = useIntl();
+  const inviteId = route.params.inviteId;
+  const {data: invite} = useSingleInvite({inviteId});
+  const acceptInvite = useAcceptInvite();
+  const rejectInvite = useRejectInvite();
+  const {setActiveProjectId} = useActiveProjectIdActions();
+  const projects = useManyProjects();
+
+  useListenToInviteStateUpdate(inviteId);
+
+  function accept() {
+    if (projects.data.length > 1) {
+      navigation.replace('AlreadyOnProject', {
```

Comment:
based on the screen name alone, took me a moment to realize that this is for when you're on an existing project and attempting to join a new project. i initially thought this is referring to some UI that is explaining that you're already on the project that you're being invited to.

think it would be helpful to rename that screen so that difference is clearer.